src/integ-test/java/com/mathworks/ci/RunMATLABTestsInteg.java

```diff
@@ -0,0 +1,263 @@
+package com.mathworks.ci;
+
+import com.gargoylesoftware.htmlunit.*;
+import com.gargoylesoftware.htmlunit.html.*;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.tasks.Builder;
+import org.junit.*;
+import hudson.EnvVars;
+import hudson.matrix.Axis;
+import hudson.matrix.AxisList;
+import hudson.matrix.Combination;
+import hudson.matrix.MatrixBuild;
+import hudson.matrix.MatrixProject;
+import hudson.matrix.MatrixRun;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
+import hudson.tasks.Builder;
+
+
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.*;
+
+
+public class RunMATLABTestsInteg {
+
+    private FreeStyleProject project;
+    private UseMatlabVersionBuildWrapper buildWrapper;
+    private RunMatlabTestsBuilder testBuilder;
+
+
+    @Rule
+    public JenkinsRule jenkins = new JenkinsRule();
+
+    @Before
+    public void testSetup() throws IOException {
+
+        this.project = jenkins.createFreeStyleProject();
```

Comment:
The test cases here are based on the FreeStyleProject configuration. Could start thinking about how to add test cases for a Pipeline project too.